### PR TITLE
fix(hooks): set sticky correctly on element change

### PR DESCRIPTION
**Pull request by marstamm**
related to https://github.com/bpmn-io/bpmn-js-properties-panel/issues/726
(see issue for why this was broken)

### CODE REVIEW

File: test/spec/hooks/useStickyIntersectionObserver.spec.js [hunk 1]
```diff
@@ -113,14 +140,30 @@ describe('hooks/userStickyIntersectionObserver', function() {
...
 
-  global.IntersectionObserver = class IntersectionObserver {
-    constructor() {}
+  const callbacks = [];
+
+  function triggerCallbacks(args) {
+    callbacks.forEach(callback => callback(args));
+  }
+
+  const MockObserver = class IntersectionObserver {
```
Comment (barmac) [inline test/spec/hooks/useStickyIntersectionObserver.spec.js:163]: `extends` is necessary if you want the mock to have IntersectionObserver's APIs.

```suggestion
  const MockObserver = class extends IntersectionObserver {
```

Comment (marstamm) [reply test/spec/hooks/useStickyIntersectionObserver.spec.js:163]: I don't think we want to extend it, rather replace it with our own implementation. But you are right that this looks wrong an could just be
```suggestion
  class MockObserver {
```

**Comment by nikku** [comment]
Try out via 

```
npx @bpmn-io/sr bpmn-io/bpmn-js-properties-panel -l bpmn-io/properties-panel#fix-sticky-headers
```

**Comment by marstamm** [comment]
Good catch @nikku. This is a z-index issue that we should fix in the headers. The FEEL marker already uses z-index 1, so the header should use something higher. I'll adjust the PR

**Review by nikku — APPROVED** [review]

**Pull request merged**